conda/plugins/shells/posix/posix_activate.py

```diff
@@ -0,0 +1,224 @@
+
+    This child class is inentionally a near-replica of the current PosixActivator class:
+    the only difference is the included _parse_and_set_args method.
+    """
+
+    pathsep_join = ":".join
+    sep = "/"
+    path_conversion = staticmethod(native_path_to_unix)
+    script_extension = ".sh"
+    tempfile_extension = None  # output to stdout
+    command_join = "\n"
+
+    unset_var_tmpl = "unset %s"
+    export_var_tmpl = "export %s='%s'"
+    set_var_tmpl = "%s='%s'"
+    run_script_tmpl = '. "%s"'
+
+    hook_source_path = join(
+        CONDA_PACKAGE_ROOT,
+        "shell",
+        "etc",
+        "profile.d",
+        "conda.sh",
+    )
+
+    def _update_prompt(self, set_vars, conda_prompt_modifier):
+        ps1 = self.environ.get("PS1", "")
+        if "POWERLINE_COMMAND" in ps1:
+            # Defer to powerline (https://github.com/powerline/powerline) if it's in use.
+            return
+        current_prompt_modifier = self.environ.get("CONDA_PROMPT_MODIFIER")
+        if current_prompt_modifier:
+            ps1 = re.sub(re.escape(current_prompt_modifier), r"", ps1)
+        # Because we're using single-quotes to set shell variables, we need to handle the
+        # proper escaping of single quotes that are already part of the string.
+        # Best solution appears to be https://stackoverflow.com/a/1250279
+        ps1 = ps1.replace("'", "'\"'\"'")
+        set_vars.update(
+            {
+                "PS1": conda_prompt_modifier + ps1,
+            }
+        )
+
+    def _hook_preamble(self) -> str:
+        result = []
+        for key, value in context.conda_exe_vars_dict.items():
+            if value is None:
+                # Using `unset_var_tmpl` would cause issues for people running
+                # with shell flag -u set (error on unset).
+                result.append(self.export_var_tmpl % (key, ""))
+            elif on_win and ("/" in value or "\\" in value):
+                result.append(f'''export {key}="$(cygpath '{value}')"''')
+            else:
+                result.append(self.export_var_tmpl % (key, value))
+        return "\n".join(result) + "\n"
+
+    # this will be in _Activate parent class logic once architecture is decided
+    def _parse_and_set_args(self, args: argparse.Namespace) -> None:
+        """
+        Set self.command to the specified command (activate, deactivate, reactivate).
+        Set context.dev if a --dev flag exists.
+        For activate, set self.env_name_or_prefix and self.stack.
+        """
+        self.command = args.command
+        context.dev = args.dev or context.dev
+
+        if self.command == "activate":
+            self.env_name_or_prefix = args.env or "base"
+
+            if args.stack is None:
+                self.stack = context.auto_stack and context.shlvl <= context.auto_stack
+            else:
+                self.stack = args.stack
+
+        return
+
+
+def get_parsed_args(argv: list[str]) -> argparse.Namespace:
```

Comment:
Since these are identical for all 3 prototypes can we move this into a new `common.py` submodule? will help make this easier to review

Reply:
`get_parsed_args` actually has a small difference across the three (the version in `posix_activate.py` sets `self.command` while the other two don't. I can look for other logic that's common across all three and extract it, though.